cpp/tests/strings/split_tests.cpp

```diff
@@ -0,0 +1,160 @@
+}
+
+TEST_F(StringsSplitTest, RSplitWhitespace)
+{
+    std::vector<const char*> h_strings{ "héllo", nullptr, "a_bc_déf", "", " a\tb ", " a\r bbb   c" };
+    cudf::test::strings_column_wrapper strings( h_strings.begin(), h_strings.end(),
+        thrust::make_transform_iterator( h_strings.begin(), [] (auto str) { return str!=nullptr; }));
+
+    cudf::strings_column_view strings_view( strings );
+    std::vector<const char*> h_expected1{ "héllo", nullptr, "a_bc_déf", nullptr, "a", "a" };
+    cudf::test::strings_column_wrapper expected1( h_expected1.begin(), h_expected1.end(),
+        thrust::make_transform_iterator( h_expected1.begin(), [] (auto str) { return str!=nullptr; }));
+    std::vector<const char*> h_expected2{ nullptr, nullptr, nullptr, nullptr, "b", "bbb" };
+    cudf::test::strings_column_wrapper expected2( h_expected2.begin(), h_expected2.end(),
+        thrust::make_transform_iterator( h_expected2.begin(), [] (auto str) { return str!=nullptr; }));
+    std::vector<const char*> h_expected3{ nullptr, nullptr, nullptr, nullptr, nullptr, "c" };
+    cudf::test::strings_column_wrapper expected3( h_expected3.begin(), h_expected3.end(),
+        thrust::make_transform_iterator( h_expected3.begin(), [] (auto str) { return str!=nullptr; }));
+    std::vector<std::unique_ptr<cudf::column>> expected_columns;
+    expected_columns.push_back(expected1.release());
+    expected_columns.push_back(expected2.release());
+    expected_columns.push_back(expected3.release());
+    auto expected = std::make_unique<cudf::experimental::table>(std::move(expected_columns));
+
+    auto results = cudf::strings::rsplit(strings_view);
+    EXPECT_TRUE( results->num_columns()==3 );
+    cudf::test::expect_tables_equal(*results,*expected);
+}
+
+TEST_F(StringsSplitTest, SplitZeroSizeStringsColumns)
+{
+    cudf::column_view zero_size_strings_column( cudf::data_type{cudf::STRING}, 0, nullptr, nullptr, 0);
+    auto results = cudf::strings::split(zero_size_strings_column);
+    EXPECT_TRUE( results->num_columns()==1 );
+    cudf::test::expect_strings_empty(results->get_column(0));
+    results = cudf::strings::rsplit(zero_size_strings_column);
+    EXPECT_TRUE( results->num_columns()==1 );
+    cudf::test::expect_strings_empty(results->get_column(0));
+}
+
+// This test specifically for https://github.com/rapidsai/custrings/issues/119
+TEST_F(StringsSplitTest, AllNullsCase)
+{
+    std::vector<const char*> h_strings{ nullptr, nullptr, nullptr };
+    cudf::test::strings_column_wrapper strings( h_strings.begin(), h_strings.end(),
+        thrust::make_transform_iterator( h_strings.begin(), [] (auto str) { return str!=nullptr; }));
+
+    auto results = cudf::strings::split(cudf::strings_column_view(strings));
+    EXPECT_TRUE( results->num_columns()==1 );
+    auto column = results->get_column(0);
+    EXPECT_TRUE( column.size()==3 );
+    EXPECT_TRUE( column.has_nulls() );
+    EXPECT_TRUE( column.null_count()==column.size() );
+}
```

Comment:
Test of error cases? Test of zero-size delimiter?

Reply:
The zero-size delimiter testcases are the whitespace ones:
https://github.com/rapidsai/cudf/blob/02937c5c31e51642a5ee3082b37c87a76b1e497c/cpp/tests/strings/split_tests.cpp#L59
and
https://github.com/rapidsai/cudf/blob/02937c5c31e51642a5ee3082b37c87a76b1e497c/cpp/tests/strings/split_tests.cpp#L109